src/DssSpell.t.sol

```diff
@@ -2211,66 +2285,69 @@ contract DssSpellTest is DSTest, DSMath {
-        assertEq(flash.daiJoin(), address(daiJoin));
-        assertEq(flash.dai(), address(dai));
-        assertEq(flash.vow(), address(vow));
-        assertEq(flash.max(), 500 * MILLION * WAD);
-        assertEq(flash.toll(), 5 * WAD / 10000);
-        assertEq(flash.maxFlashLoan(address(dai)), 500 * MILLION * WAD);
-        assertEq(flash.flashFee(address(dai), 1 * MILLION * WAD), 500 * WAD); // 500 DAI fee on a 1M loan
-        flash.flashLoan(address(this), address(dai), 1 * MILLION * WAD, "");
-        flash.vatDaiFlashLoan(address(this), 1 * MILLION * RAD, "");
-        assertEq(vat.sin(address(flash)), 0);
-        assertEq(vat.dai(address(flash)), 1000 * RAD);
-        flash.accrue();
-        assertEq(vat.dai(address(flash)), 0);
-        assertEq(vat.dai(address(vow)), vowDai + 1000 * RAD);
-    }
+        assertEq(vat.can(address(pauseProxy), address(daiJoin)), 1);
+
+        assertEq(
+            vat.sin(address(vow)) - prevSin,
+            ( amountGro
+            + amountMkt
+            + amountGov
+            + amountRwf
+            + amountRisk
+            + amountPe
+            + amountOra
+            + amountOraEr
+            ) * RAD
+        );
+        assertEq(vat.sin(address(vow)) - prevSin, amountTotal * RAD);
+        assertEq(dai.balanceOf(GRO_MULTISIG)    - prevDaiGro, amountGro * WAD);
+        assertEq(dai.balanceOf(MKT_MULTISIG)    - prevDaiMkt, amountMkt * WAD);
+        assertEq(dai.balanceOf(GOV_MULTISIG)    - prevDaiGov, amountGov * WAD);
+        assertEq(dai.balanceOf(RWF_MULTISIG)    - prevDaiRwf, amountRwf * WAD);
+        assertEq(dai.balanceOf(RISK_CU_EOA)     - prevDaiRisk, amountRisk * WAD);
+        assertEq(dai.balanceOf(PE_MULTISIG)     - prevDaiPe, amountPe * WAD);
+        assertEq(dai.balanceOf(ORA_MULTISIG)    - prevDaiOra, amountOra * WAD);
+        assertEq(dai.balanceOf(ORA_ER_MULTISIG) - prevDaiOraEr, amountOraEr * WAD);
 
-    function onFlashLoan(
-        address initiator,
-        address token,
-        uint256 amount,
-        uint256 fee,
-        bytes calldata
-    ) external returns (bytes32) {
-        assertEq(initiator, address(this));   
-        assertEq(token, address(dai));
-        assertEq(amount, 1 * MILLION * WAD);
-        assertEq(fee, 500 * WAD);
-
-        dai.approve(msg.sender, 1_000_500 * WAD);
-
-        return keccak256("ERC3156FlashBorrower.onFlashLoan");
     }
 
-    function onVatDaiFlashLoan(
-        address initiator,
-        uint256 amount,
-        uint256 fee,
-        bytes calldata
-    ) external returns (bytes32) {
-        assertEq(initiator, address(this));   
-        assertEq(amount, 1 * MILLION * RAD);
-        assertEq(fee, 500 * RAD);
-
-        dai.approve(address(daiJoin), 500 * WAD);
-        daiJoin.join(address(this), 500 * WAD);
-        vat.move(address(this), msg.sender, 1_000_500 * RAD);
-
-        return keccak256("VatDaiFlashBorrower.onVatDaiFlashLoan");
-    }
+    function testClawBackMultisigFunds() public {
```

Comment:
Should probably also assert on the size of the allowance--this confirms that a single DAI can be reclaimed, but if that uses up the whole allowance, it's not very meaningful in practice.

Reply:
added at 89804b77e736b353e09b7930c2c2b3a3e1579afd